packages/vue-app/template/client.js

```diff
@@ -279,6 +279,7 @@ async function render(to, from, next) {
   // Get route's matched components
   const matches = []
   const Components = getMatchedComponents(to, matches)
+  const instances = getMatchedComponentsInstances(to)
```

Comment:
Could this be moved to the watchQuery `else if` (eg using memoization)?

The need for `instances` is for such a specific use-case only that we shouldnt always spent those cpu cycles retrieving them when there is a reasonable chance we dont need them

Reply:
Done.